front/components/data_source/DataSourceEdition.tsx

```diff
@@ -0,0 +1,436 @@
+import {
+  Avatar,
+  Button,
+  CloudArrowLeftRightIcon,
+  Dialog,
+  Icon,
+  LockIcon,
+  Modal,
+  Page,
+} from "@dust-tt/sparkle";
+import type {
+  ConnectorProvider,
+  ConnectorType,
+  DataSourceIntegration,
+  DataSourceType,
+  EditedByUser,
+  UpdateConnectorRequestBody,
+  UserType,
+  WorkspaceType,
+} from "@dust-tt/types";
+import { CONNECTOR_TYPE_TO_MISMATCH_ERROR } from "@dust-tt/types";
+import { InformationCircleIcon } from "@heroicons/react/20/solid";
+import type { NextRouter } from "next/router";
+import React, { useContext, useState } from "react";
+
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { CONNECTOR_CONFIGURATIONS } from "@app/lib/connector_providers";
+import { formatTimestampToFriendlyDate } from "@app/lib/utils";
+import type { PostManagedDataSourceRequestBody } from "@app/pages/api/w/[wId]/data_sources/managed";
+import { setupConnection } from "@app/pages/w/[wId]/builder/data-sources/managed";
+
+interface DataSourceEditionModalProps {
+  isOpen: boolean;
+  owner: WorkspaceType;
+  connectorProvider: ConnectorProvider | null | undefined;
+  dataSourceIntegration: DataSourceIntegration | DataSourceType | null;
+  onClose: () => void;
+  router: NextRouter;
+  dustClientFacingUrl: string;
+  user: UserType;
+  setIsRequestDataSourceModalOpen?: (show: boolean) => void;
+  setDataSourceIntegrations?: (
+    integrations: (prev: DataSourceIntegration[]) => DataSourceIntegration[]
+  ) => void;
+  setIsLoadingByProvider?: (
+    providers: (
+      prev: Record<ConnectorProvider, boolean | undefined>
+    ) => Record<ConnectorProvider, boolean | undefined>
+  ) => void;
+}
+
+const REDIRECT_TO_EDIT_PERMISSIONS = [
```

Comment:
Can we use proper typing here so the linter will fail if we add a new connector. We can probably do `Omit<..., ...>`

Reply:
I ended up using `Extract`, seemed convenient in this case 😇